src/public/modules/forms/admin/controllers/view-responses.client.controller.js

```diff
@@ -244,11 +246,102 @@ function ViewResponsesController(
     })
   }
 
-  vm.downloadAllAttachments = function () {
+  vm.confirmSubmissionCountsBeforeDownload = function () {
+    let params = {
+      formId: vm.myform._id,
+      formTitle: vm.myform.title,
+    }
+    if (vm.datePicker.date.startDate && vm.datePicker.date.endDate) {
+      params.startDate = moment(new Date(vm.datePicker.date.startDate)).format(
+        'YYYY-MM-DD',
+      )
+      params.endDate = moment(new Date(vm.datePicker.date.endDate)).format(
+        'YYYY-MM-DD',
+      )
+    }
+    Submissions.count(params).then((responsesCount) => {
+      vm.downloadAttachmentResponsesCount = responsesCount
+      vm.confirmDownloadModal = $uibModal.open({
+        backdrop: 'static',
+        templateUrl:
+          'modules/forms/admin/views/download-all-attachments.client.modal.html',
+        scope: $scope,
```

Comment:
can we explicitly pass only what's required instead of the entire scope?

Reply:
Given that the modal only displays one number and calls two functions (either to continue the download or to dismiss the modal), passing in a new scope would mean creating a new controller and I am not sure that is better since that creates a bunch of boilerplate that we don't need?

Reply:
yep, creating a new scope is too much, but actually that's not necessary. there is a canonical solution by using the `resolve` parameter in `$uibModal.open` to pass the number into the view.

we can also leave `vm.confirmDownloadModal.close()` and `vm.downloadMultipleSubmissionAttachments()` out of the modal by using `scope.$close` to close on success, or `scope.$dismiss` to close and cancel from inside (as per the [docs](https://angular-ui.github.io/bootstrap/)).

Probably something like this:

```html
// download-all-attachments.client.modal.html
  <div class="modal-footer">
    <button
      type="button"
      class="btn btn-primary"
      ng-click="vm.$close()"
    >
      Yes, download attachments
    </button>
    <button
      type="button"
      class="btn btn-default"
      ng-click="vm.$dismiss()"
    >
      Close
    </button>
  </div>```
```
```javascript
// view-responses.client.controller.js
$uibModal.open(...).result
  .then(() => downloadMultipleSubmissionAttachments())
  .catch(() => console.log('user decided to cancel')
```

this [plunkr](http://plnkr.co/edit/yB3k8e3R3ZLQFQ6sfLYW) demonstrates the use of `closed`.